core/aws_ddk_core/stages/athena_sql.py

```diff
@@ -135,3 +150,16 @@ def __init__(
             state_machine_failed_executions_alarm_threshold=state_machine_failed_executions_alarm_threshold,
             state_machine_failed_executions_alarm_evaluation_periods=state_machine_failed_executions_alarm_evaluation_periods,  # noqa
         )
+
+    def get_targets(self) -> Optional[List[IRuleTarget]]:
+        return [
+            SfnStateMachine(
+                self.state_machine,
+                input=RuleTargetInput.from_object(self._state_machine_input),
+            )
+            if self._query_string
+            else SfnStateMachine(
+                self.state_machine,
+                input=RuleTargetInput.from_event_path(self._event_bridge_event_path),
```

Comment:
It looks like only the state machine input changes, could it be simplified into `SfnStateMachine` with a conditional `input=RuleTargetInput.from_object... if ... else RuleTargetInput.from_event_path...`?

Reply:
+1

Reply:
Thanks, Done